pubsub/topic.go

```diff
@@ -585,26 +593,51 @@ func (t *Topic) Publish(ctx context.Context, msg *Message) *PublishResult {
 		Attributes:  msg.Attributes,
 		OrderingKey: msg.OrderingKey,
 	})
+	span.SetAttributes(semconv.MessagingMessagePayloadSizeBytesKey.Int(msgSize))
 
 	t.initBundler()
 	t.mu.RLock()
 	defer t.mu.RUnlock()
 	// TODO(aboulhosn) [from bcmills] consider changing the semantics of bundler to perform this logic so we don't have to do it here
 	if t.stopped {
 		ipubsub.SetPublishResult(r, "", ErrTopicStopped)
+		spanRecordError(span, ErrTopicStopped)
 		return r
 	}
 
+	ctx2, fcSpan := tracer().Start(ctx, publishFlowControlSpanName)
```

Comment:
Please bear with me as I push my way through not knowing Go :D

Should the new ctx2 above be used for flowController.acquire() and such? If not, can you explain what's going on there with the contexts?

Reply:
Yeah that's probably a good idea so it reads better though in reality it doesn't make much of a difference. Contexts are just used to pass around information and the flow controller doesn't really need to know about the additional otel span bits the `ctx2` got augmented with.